src/codemagic/apple/resources/build.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from datetime import datetime
+
+from .enums import BuildProcessingState
+from .resource import Resource
+from .resource import DictSerializable
+from .resource import Relationship
+
+
+class Build(Resource):
+    """
+    https://developer.apple.com/documentation/appstoreconnectapi/build
+    """
+
+    @dataclass
+    class Attributes(Resource.Attributes):
+        expired: bool
+        iconAssetToken: ImageAsset
+        minOsVersion: str
+        processingState: BuildProcessingState
+        version: str
+        usesNonExemptEncryption: bool
+        uploadedDate: datetime
+        expirationDate: datetime
+
+        def __post_init__(self):
+            if isinstance(self.processingState, str):
+                self.processingState = BuildProcessingState(self.processingState)
+            if isinstance(self.uploadedDate, str):
+                self.uploadedDate = Resource.from_iso_8601(self.uploadedDate)
+            if isinstance(self.expirationDate, str):
+                self.expirationDate = Resource.from_iso_8601(self.expirationDate)
+
+    @dataclass
+    class Relationships(Resource.Relationships):
+        app: Relationship
+        appEncryptionDeclaration: Relationship
+        individualTesters: Relationship
+        preReleaseVersion: Relationship
+        betaBuildLocalizations: Relationship
+        buildBetaDetail: Relationship
+        betaAppReviewSubmission: Relationship
+        appStoreVersion: Relationship
+        icons: Relationship
+        betaGroups: Relationship
```

Comment:
they always come in build's response, it would already fail if they weren't present in at least one build.
Sure can make them optional with default `None` and redefine `asdict` function to not return `None` fields

Reply:
Since they are not defined in API doc, I'd maybe define them like
```suggestion
        betaGroups: Optional[Relationship] = None
```
Of course we'd need to make sure that `asdict` and any other method would remain functional.

Reply:
My main concern is that something along the lines of the following would happen:
```python
>>> @dataclass
... class MyDataclass:
...     i: int
...     s: str
>>> MyDataclass(i=1)
Traceback (most recent call last):
  File "<input>", line 5, in <module>
TypeError: __init__() missing 1 required positional argument: 's'
```

Reply:
fixed [here](7cf35a0d94ec2496fb6053f99fda415a08b96a6d)